packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1903,9 +1916,12 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
             .forEach((Selectable target) => dispatchSelectionEventToChild(target, const ClearSelectionEvent()));
           currentSelectionStartIndex = currentSelectionEndIndex = index;
         }
-        return SelectionResult.end;
+        return lastSelectionResult;
```

Comment:
oops, I forgot to press send for my previous review. I will repaste here

This doesn't look right. consider these cases, assuming 5 child, skip means the selectword point is not in the rect

1.
selectable 1 -> skipped (continue)
selectable 2 -> next (continue)
selectable 3 -> next (continue)
selectable 4 -> prev (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 3 or 4
should return SelectionResult.end

2.
selectable 1 -> skipped (continue)
selectable 2 -> prev (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end

3.
selectable 1 -> prev (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : null
should return SelectionResult.prev

4.
selectable 1 -> skip (continue)
selectable 2 -> next (continue)
selectable 3 -> end (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 3
should return SelectionResult.end

5.
selectable 1 -> skip (continue)
selectable 2 -> next (continue)
selectable 3 -> skip (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end

6.
selectable 1 -> skip (continue)
selectable 2 -> skip (continue)
selectable 3 -> skip (continue)
selectable 4 -> next (continue)
selectable 5 -> next (at the end)

currentSelectionStartIndex/currentSelectionEndIndex : null
should return SelectionResult.next

7.
selectable 1 -> skip (continue)
selectable 2 -> skip (continue)
selectable 3 -> skip (continue)
selectable 4 -> skip (continue)
selectable 5 -> skip (at the end)

currentSelectionStartIndex/currentSelectionEndIndex : null
should probably return SelectionResult.next

Reply:
You're right, think it should be fixed now. But I'm a bit unsure about this case

```
5.
selectable 1 -> skip (continue)
selectable 2 -> next (continue)
selectable 3 -> skip (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end
```

Why does it stop at the last skip?

Reply:
This assume all selectable after 3 are skipped. If not, it is likely a bug in sorting screen order. I am actually not sure whether this should return next or end, but I can't think of a case returning end would be a problem.

Reply:
I see, so what this case means is that in a given list of `selectables` a `next` was reported but it was not followed by a `prev`/or `end` that would indicate the word is inside of the list of `selectables`? Why in this case then do we still have to set the currentSelectionStartIndex/currentSelectionEndIndex : 2, when that might not be where the word is? Thank you for explaining.

Reply:
If we return `end`, we have to give a currentSelectionStartIndex/currentSelectionEndIndex. If we return `next`, we should not set currentSelectionStartIndex/currentSelectionEndIndex.

After I thought about it more, I think we should return `end` to short-circuit the logic. Otherwise the `select world` logic would need to go through every selectable in the subtree. to figure out whether the position has hit any selectables.

There are two mechanism in play here: the `rect` and `selection result`. In theory, we don't need `rect`, and we can completely rely on `selection result`. The `rect` is here so that we can trim the search tree. If we were to return `next` in this kind of un-decided case, we might as well remove the `rect` check completely.

That also reminding me I was wrong on a previous case. These two case should be the following

5.
selectable 1 -> skip (continue)
selectable 2 -> next (continue)
selectable 3 -> skip (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end

7
selectable 1 -> skip (continue)
selectable 2 -> skip (continue)
selectable 3 -> skip (continue)
selectable 4 -> skip (continue)
selectable 5 -> skip (at the end)

currentSelectionStartIndex/currentSelectionEndIndex : null
should return SelectionResult.end

Reply:
basically, 

if the first selection return prev, this method return prev
if the [n...length-1] return next, this method return next
else return SelectionResult.end.